content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,38 @@
+
+This is what we call a code intelligence platform, and compared to code search, it's more helpful, more times per day. **A code intelligence platform helps devs learn and understand information about their code so they can operate more effectively, with a deeper understanding of their codebase and how it works.** 
+
+A code intelligence platform should persist data beyond the code itself and make that data accessible. Here are some examples of what we see the platform doing in the future:
+
+- Trying to figure out the best way to call a function? Code search could show you a semi-exhaustive list of call sites. Code intelligence is also being able to see how the code owners of that function themselves have recently been calling that function.
+- Need to debug an issue in code? Code search could help you navigate within the same project to understand it. Code intelligence is also being able to see what changed and why, and trace calls across projects and dependencies/dependents.
+- Want to fix or improve something across your entire codebase? Code search might help you find where needs fixing. Code intelligence is also being able to push up a quick fix everywhere, track its progress, and monitor for regressions. Sourcegraph does this today.
+- Need to see how your team's API is being used? Code search could show you a list of file locations. Code intelligence is also seeing which other teams are using it and the trend of usage over time.
+
+A code intelligence platform includes code search *and* information about your code from so many other tools, etc., that you rely on: code ownership (and mapping of that to teams, on-call, etc.), code history and reviews, the dependency graph, runtime service/API usage, code and security scanners, deployment status, CI pipeline output, etc. Sourcegraph is committed to being universal so you can get full code intelligence *and* keep using the best tools in each category.
+
+Today, we're announcing Sourcegraph 4.0, the code intelligence platform for devs to unblock themselves and code better. We've spent 9 years building the underlying foundation of code search to be fast, secure, and scalable across ~10^11 lines of code for the world's most demanding software teams, including 4 of 5 FAANG companies and 4 of the top 10 U.S. banks. 
+
+Sourcegraph has lots to do to build our future vision of a code intelligence platform, and we're already on our way. Our newest developments—specifically, Code Insights and Batch Changes—have solidly moved Sourcegraph _beyond_ code search, and today, we're ready to call Sourcegraph a code intelligence platform.
+
+Sourcegraph 4.0 is available on Cloud, custom deployments, and self-hosted. If you're new to Sourcegraph,  Sourcegraph Cloud (secure & dedicated instances) is the easiest way to get started. You can get a [30-day trial here](www.signup.sourcegraph.com).
```

Comment:
I think we should also introduce Cloud a bit more here as something new.

Reply:
```suggestion
Sourcegraph 4.0 is available to all users now. It's also available on our newest service, Sourcegraph Cloud, our single-tenant Cloud solution that is [now generally available](about.sourcegraph.com/blog/enterprise-cloud). Sourcegraph Cloud provides isolated, secure instances for every customer.

If you're already using Sourcegraph, you can [upgrade to 4.0 today](https://docs.sourcegraph.com/admin/updates). If you're new to Sourcegraph, Sourcegraph Cloud is the easiest way to get started. You can get a [30-day trial here](www.signup.sourcegraph.com).
```

Reply:
We can link to the Cloud blog in here as well.